File: sol/solution/8 Kyu/578c1e2edaa01a9a02000b7f/Solution.rb
=begin
Solution
=end
def alias_gen(first_name, surname)
  f = first_name[0].upcase
  s = surname[0].upcase
  unless f =~ /[A-Z]/ && s =~ /[A-Z]/
    return "Your name must start with a letter from A - Z."
  end
  "#{FIRST_NAME[f]} #{SURNAME[s]}"
end





=begin
Sample Tests
=end
def dotest(first_name, surname, expected)
  it "first_name = #{first_name.inspect}, surname = #{surname.inspect}" do
    expect(alias_gen(first_name, surname)).to eq expected
  end
end

describe "Fixed tests" do
  dotest("Mike", "Millington", "Malware Mike")
  dotest("Fahima", "Tash", "Function T-Rex")
  dotest("Daisy", "Petrovic", "Data Payload")
  dotest("Barny", "White", "Beta Worm")
  dotest("Hank", "Kutz", "Half-life Killer")
  dotest("johnny", "Tomato", "Java T-Rex")
  dotest("7393424", "Anumbha", "Your name must start with a letter from A - Z.")
  dotest("Anuddanumbha", "23200", "Your name must start with a letter from A - Z.")
end

File: sol/solution/8 Kyu/578c1e2edaa01a9a02000b7f/Solution.js
/**
 * Solution
 */
function aliasGen(first, last){
  const f = first[0].toUpperCase();
  const l = last[0].toUpperCase();
  if(!/^[A-Z]$/.test(f) || !/^[A-Z]$/.test(l)){
    return "Your name must start with a letter from A - Z.";
  }
  return `${firstName[f]} ${surname[l]}`;
}





/**
 * Sample Tests
 */
const { assert } = require('chai');

describe("Tests", () => {
  it("test", () => {
    assert.strictEqual(aliasGen("Mike", "Millington"), "Malware Mike");
    assert.strictEqual(aliasGen("Fahima", "Tash"), "Function T-Rex");
    assert.strictEqual(aliasGen("Daisy", "Petrovic"), "Data Payload");
    assert.strictEqual(aliasGen("7393424", "Anumbha"), "Your name must start with a letter from A - Z.");
    assert.strictEqual(aliasGen("Anuddanumbha", "23200"), "Your name must start with a letter from A - Z.");
  });
});
